File: seeker/snippet/to_lower.go
//date: 2024-06-11T16:44:51Z
//url: https://api.github.com/gists/2dede5a7a0ae593a1a0da14d10d5ebfb
//owner: https://api.github.com/users/mdigger

// FirstRuneToLower converts first rune to lower case if necessary.
func FirstRuneToLower(str string) string {
	if str == "" {
		return str
	}

	r, size := utf8.DecodeRuneInString(str)

	if !unicode.IsUpper(r) {
		return str
	}

	buf := &stringBuilder{}
	buf.WriteRune(unicode.ToLower(r))
	buf.WriteString(str[size:])
	return buf.String()
}

File: seeker/snippet/tolower.py
#date: 2025-10-31T17:03:26Z
#url: https://api.github.com/gists/6b0fb11f5f409efbddd3f76703b582e6
#owner: https://api.github.com/users/idolpx

#!/usr/local/bin/python3
import os

list = []

dirlist = [os.getcwd() + '/']

# Build recursive list for dirs and files
while len(dirlist) > 0:
    for (dirpath, dirnames, filenames) in os.walk(dirlist.pop()):
        dirlist.extend(dirnames)
        list.extend( map(lambda n: os.path.join(*n), zip([dirpath] * len(dirnames), dirnames)) )
        list.extend( map(lambda n: os.path.join(*n), zip([dirpath] * len(filenames), filenames)) )

# Iterate through list backwards and rename files and folders
for x in range(len(list) -1, -1, -1):
    item = list[x]
    print(f'{item} >>> {item.lower()}')
    os.rename(item, item.lower())